src/lua/LuaGame.cpp

```diff
@@ -776,6 +816,7 @@ void LuaGame::Register()
 
 		{ "SwitchView", l_game_switch_view },
 		{ "CurrentView", l_game_current_view },
+		{ "PreviousView", l_game_previous_view },
```

Comment:
I would generally prefer to add additional events fired by a View when it is made (in)active than introduce this new method - though if pushing/popping the InputFrame is implemented according to https://github.com/pioneerspacesim/pioneer/pull/5958#discussion_r1844271088 neither should be necessary.

Reply:
Hmm.. ok, getting very mixed signals (pun not intended) here..  in another PR you wanted me to remove events and use function calls. Here you're saying I should be using events.

What paradigm is this codebase supposed to use?!

Reply:
Sorry about the confusion - there are three (now four) possible states to implement the functionality of popping the input frame:
1. The simplest way would be to statically call `RemoveFromStack` every time `onViewChanged` is triggered, regardless of if the input frame is active. (I can add an `Input.HasInputFrame(frame)` method if you're concerned about the performance here.)
2. Pass the name of the previous frame as a parameter to the `onViewChanged` event.
3. If option 1 or 2 is not chosen, I'd prefer PiGuiView queue an `onViewActivated` / `onViewDeactivated` event.
4. The currently-implemented method of `Game.PreviousView`, which adds an additional API call and duplicates the view name lookup code.

I had originally suggested option 1 in https://github.com/pioneerspacesim/pioneer/pull/5958#discussion_r1844271088, with this comment expressing my desire to use option 3 instead of 4 if option 1 was not chosen. Upon reflection, I think option 2 would be the simplest possible way to achieve this, while still being performant.

Reply:
At this point - just pick whatever you want to implement; I've nitpicked the PR to death and you're more than welcome to say "not going to bother" to nitpicks :)

Reply:
Lol, I actual thought of "2" while I was reverting..   I'll leave it as-is for now, but adding "2" at some point would be a good addition I think. That'd be the most performant and least invasive.